docs/TROUBLE.md

```diff
@@ -0,0 +1,175 @@
+
+- Be sure you have flashed the correct [firmware file](https://github.com/olliw42/mLRS#firmware-flashing).  Files which start with tx are only for modules functioning in the Tx role and files which start with rx are only for modules functioning in the receiver role.  Also, pay attention to the variants for different use cases and hardware configurations.  For example, there are 3 separate files for different hardware and use case options for the MatekSys mR900-30 when used as a Tx module and a 4th file for when the same hardware is used as a Rx.  A module with rx firmware flashed won't connect with any other module with rx firmware flashed and tx firmware won't connect with tx firmware.
+
+## 4. Confirm MAVLink message flow from flight controller to receiver and Tx module
+
+In EdgeTX/OpenTX go to MDL->TELEMETRY and select "Discover new sensors".  Then, scroll up using the scroll wheel and you should find a sensor named FM among several other sensors.  It should have a star flashing next to it indicating it is updating about once per second.  See the [CRSF](CRSF.md] documentation page for background info.
+
+Above, you have already confirmed the radio link and communication between the Tx module and OpenTX/EdgeTX.  So, if you don't see the FM sensor updating, this very likely indicates that the HEARTBEAT MAVLink message which is sent once per second by ArduPilot on all MAVLink interfaces is not reaching the Rx.
+
+- Check your serial port wiring between the flight controller and the receiver to ensure you have connected serial rx on one board to serial tx on the other board in both directions and ground to ground.
+
+- Use the mLRS LUA script or another configuration method to check your [mLRS receiver setup](CRSF.md#mlrs-rx-module-setup) to be sure MAVLink is selected for the Rx Ser Link Mode and that [Rx Ser Baudrate](PARAMETERS.md#rx-ser-baudrate) is correct and matches what you set in the following bullet.
+
+- Use a USB connected ground control station to check your ArduPilot serial [configuration](CRSF.md#ardupilot-setup) to be sure the correct SERIALx\_PROTOCOL is set to MAVLink2 and that the SERIALx\_BAUD matches what you set the receiver to use in the previous bullet.
+
+- If it is still not working, use a volt meter or oscilloscope to verify that both serial port wires between the mLRS receiver and flight controller have a voltage with respect to ground that is constantly changing.  If either wire has no changing signal, you have probably still missed something above.
+
+## 5. Confirm MAVLink message flow from Tx module to ground station
+
+Start your ground station software and connect to your Tx module by whatever method you have chosen.  You should see or hear the flight mode announced.  If parameter download also completes successfully, you can skip the next two topics
```

Comment:
- What is announcing the flight mode?  I've never heard this?
- Nitpick 'If the parameter download...'

Reply:
QGC speaks the flight mode whenever it changes and when it is first received.  I don't know if MP can be configured to speak.  In my test, MP just displays it.  Do I need make this more clear?

Reply:
May help to just put the GCS parentheses, so 'You should see (MP) or hear the flight mode announced (QGC).'

Reply:
Yes, that's better.  I also removed the word announced.